src/core/raster/qgssinglebandgrayrenderer.cpp

```diff
@@ -218,3 +218,154 @@ QList<int> QgsSingleBandGrayRenderer::usesBands() const
+
+        // if minimum range differ from the real minimum => set is in exported SLD vendor option
+        if ( contrastEnhancement()->minimumValue() != myRasterBandStats.minimumValue )
+        {
+          // look for VendorOption tag to look for that with minValue attribute
+          QDomNodeList elements = contrastEnhancementElem.elementsByTagName( QStringLiteral( "sld:VendorOption" ) );
+          for ( int i = 0; i < elements.size(); ++i )
+          {
+            QDomElement vendorOption = elements.at(i).toElement();
+            if ( vendorOption.attribute( QStringLiteral( "name" ) ) != QStringLiteral( "minValue" ))
+              continue;
+
+            // remove old value and add the new one
+            vendorOption.removeChild(vendorOption.firstChild());
+            vendorOption.appendChild( doc.createTextNode( QString::number( myRasterBandStats.minimumValue ) ) );
+          }
+        }
+        break;
+      }
+      case ( QgsContrastEnhancement::NoEnhancement ):
+      case ( QgsContrastEnhancement::StretchToMinimumMaximum ):
+      default:
+        break;
+    }
+
+    channelElem.appendChild( contrastEnhancementElem );
+  }
+
+  // for each color set a ColorMapEntry tag nested into "sld:ColorMap" tag
+  // e.g. <ColorMapEntry color="#EEBE2F" quantity="-300" label="label" opacity="0"/>
+  QList< QPair< QString, QColor > > classes;
+  legendSymbologyItems( classes );
+
+  // add ColorMap tag
+  QDomElement colorMapElem = doc.createElement( QStringLiteral( "sld:ColorMap" ) );
+  rasterSymbolizerElem.appendChild( colorMapElem );
+
+  // TODO: add clip intervals basing on real min/max without trigger
+  // min/max calculation again that can takes a lot for remote or big images
+  //
+  // contrast enhancement against a color map can be SLD simulated playing with ColorMapEntryies
+  // each ContrastEnhancementAlgorithm need a specfic management.
+  // set type of ColorMap ramp [ramp, intervals, values]
+  // basing on interpolation algorithm of the raster shader
+  QList< QPair< QString, QColor > > colorMapping(classes);
+  switch ( contrastEnhancement()->contrastEnhancementAlgorithm() )
+  {
+    case ( QgsContrastEnhancement::StretchAndClipToMinimumMaximum ):
+    case ( QgsContrastEnhancement::ClipToMinimumMaximum ):
+    {
+      QString lowValue = classes[0].first;
+      QColor lowColor = classes[0].second;
+      lowColor.setAlpha(0);
+      QString highValue = classes[1].first;
+      QColor highColor = classes[1].second;
+      highColor.setAlpha(0);
+
+      colorMapping.prepend( QPair< QString, QColor >( lowValue, lowColor));
+      colorMapping.append( QPair< QString, QColor >( highValue, highColor));
+      break;
+    }
+    case ( QgsContrastEnhancement::NoEnhancement ):
+      break;
+    case ( QgsContrastEnhancement::StretchToMinimumMaximum ):
+      // use default
+    default:
```

Comment:
Remove default clause so we get warnings if new contrast methods are added in future

Reply:
removed all default I introduced